src/node/snapshotter.h

```diff
@@ -63,14 +63,23 @@ namespace ccf
     // committed
     std::deque<SnapshotInfo> pending_snapshots;
 
+    // Initial snapshot index
+    static constexpr consensus::Index initial_snapshot_idx = 0;
+
     // Index at which the lastest snapshot was generated
     consensus::Index last_snapshot_idx = 0;
 
     // Used to suspend snapshot generation during public recovery
     bool snapshot_generation_enabled = true;
 
     // Indices at which a snapshot will be next generated
-    std::deque<consensus::Index> next_snapshot_indices;
+    typedef struct
+    {
+      consensus::Index idx;
+      bool forced;
+    } snapshot_indices_entry;
```

Comment:
```suggestion
    struct SnapshotEntries
    {
      consensus::Index idx;
      bool forced;
    };
```

The comment above will need updating too.